client/src/containers/ProtectedRoute/ProtectedRoute.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Route, Redirect, withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+/**
+ * Wraps React Router's Route component with some permissions checking, allowing routes
+ * to be limited to certain user groups. If `oneOf` is not specified, then it simply
+ * checks that the user is logged into an account of any type
+ */
+class ProtectedRoute extends React.Component {
+
+    static mapStateToProps(state) {
+        return {session: state.session};
+    }
+
+    render() {
+        const { session, oneOf, component: Component, path, history } = this.props;
+
+        // Wait until we've finished getting the user session before making any redirects
+        if (session.pending) {
+            return (<></>);
+        }
+
+        const noAccessRedirect = (<Redirect to={{
+            pathname: '/',
+            state: {alert: 'You do not have sufficient privileges to view that page.'}
```

Comment:
I think the message here should be `Oopsie woopsie! You made a mistake! You can't view that page silly! :) :) :)`.